ed25519/suite.go

```diff
@@ -37,8 +42,32 @@ func (s *suiteEd25519) New(t reflect.Type) interface{} {
 	return abstract.SuiteNew(s, t)
 }
 
+// NewKey returns a formatted Ed25519 key (avoiding subgroup attack by requiring
+// it to be a multiple of 8)
+func (s *suiteEd25519) NewKey(stream cipher.Stream) abstract.Secret {
+	scalar, _ := NewEd25519Scalar(stream)
+	nint := nist.NewInt(0, &primeOrder.V)
+	nint.Bo = nist.LittleEndian
+	nint.SetBytes(scalar)
+	return nint
+}
+
 // Ciphersuite based on AES-128, SHA-256, and the Ed25519 curve.
 func NewAES128SHA256Ed25519(fullGroup bool) abstract.Suite {
 	suite := new(suiteEd25519)
 	return suite
 }
+
+// NewEd25519Scalar returns the scalar derived from the stream and the right
+// half of the hash during the derivation of the scalar.
+func NewEd25519Scalar(stream cipher.Stream) ([]byte, []byte) {
```

Comment:
Agreed (or just make it private to the module).

Also, no need for function names in this module to contain "Ed25519" when they're in a module called "ed25519"!